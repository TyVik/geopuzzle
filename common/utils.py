import string

from django.utils.crypto import get_random_string


def random_string():
    return get_random_string(12, string.ascii_lowercase + string.digits)
