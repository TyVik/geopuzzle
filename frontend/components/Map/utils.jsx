/* global google */


function decodePolygon(polygon) {
  return polygon.map(polygon => (google.maps.geometry.encoding.decodePath(polygon)));
}

export { decodePolygon };
