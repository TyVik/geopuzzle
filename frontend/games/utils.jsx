import messages from "../locale/messages";

const prepareInfobox = (json) => {
  if (json.area) {
    json.area = `${Number(json.area).toLocaleString()} ${messages[window.__LANGUAGE__].km2}`;
  }
  if (json.population) {
    json.population = Number(json.population).toLocaleString();
  }
  ['depth', 'length', 'width', 'height'].forEach(item => {
    if (json[item]) {
      json[item] = `${Number(json[item]).toLocaleString()} ${messages[window.__LANGUAGE__].m}`;
    }
  });
  return json;
};

function shuffle(a) {
  for (let i = a.length; i; i--) {
    let j = Math.floor(Math.random() * i);
    [a[i - 1], a[j]] = [a[j], a[i - 1]];
  }
  return a;
}


export {prepareInfobox, shuffle};