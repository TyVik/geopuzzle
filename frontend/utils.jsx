'use strict';
import 'whatwg-fetch';
import Cookies from 'js-cookie';

import localization from "./localization";

function moveTo(paths, from, to) {
    let newPoints =[];
    let newPaths = [];

    for (let p = 0; p < paths.length; p++) {
        let path = paths[p];
        newPoints.push([]);

        for (let i = 0; i < path.length; i++) {
            newPoints[newPoints.length-1].push({
                heading: google.maps.geometry.spherical.computeHeading(from, path[i]),
                distance: google.maps.geometry.spherical.computeDistanceBetween(from, path[i])
            });
        }
    }

    for (let j = 0, jl = newPoints.length; j < jl; j++) {
        let shapeCoords = [],
            relativePoint = newPoints[j];
        for (let k = 0, kl = relativePoint.length; k < kl; k++) {
            shapeCoords.push(google.maps.geometry.spherical.computeOffset(
                to,
                relativePoint[k].distance,
                relativePoint[k].heading
            ));
        }
        newPaths.push(shapeCoords);
    }
    return newPaths;
}


function decodePolygon(polygon) {
    return polygon.map(polygon => (google.maps.geometry.encoding.decodePath(polygon)));
}

const prepareInfobox = (json) => {
    if (json.area) {
        json.area = Number(json.area).toLocaleString() + ' ' + localization.km2;
    }
    if (json.population) {
        json.population = Number(json.population).toLocaleString();
    }
    return json;
};

function shuffle(a) {
    for (let i = a.length; i; i--) {
        let j = Math.floor(Math.random() * i);
        [a[i - 1], a[j]] = [a[j], a[i - 1]];
    }
    return a;
}


const CSRFfetch = (url, options) => {
    let headers = options.headers || new Headers();
    headers.append('X-CSRFTOKEN', Cookies.get('csrftoken'));
    return fetch(url, {...options, headers: headers, credentials: 'same-origin'});
};


export {moveTo, decodePolygon, prepareInfobox, shuffle, CSRFfetch};
