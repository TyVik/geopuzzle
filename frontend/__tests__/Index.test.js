'use strict';
import React from "react";
import {createComponentWithIntl} from "./utils";
import Index from "../index/index";


describe('shallow <Index /> components', () => {
  let id = 0;
  let game = name => {
    id += 1;
    return {id: id, image: `${name}.jpg`, name: name, url: `/puzzle/${name}/`};
  };

  beforeAll(() => {
    let world = ["hard", "easy"].map(item => game(item));
    let parts = ["europe", "america", "asia", "africa"].map(item => game(item));
    window.__GAMES__ = [
      {items: {world: world, parts: parts}, name: "puzzle", link: "puzzle_map", caption: "Puzzle"},
      {items: {world: world, parts: parts}, name: "quiz", link: "quiz_map", caption: "Quiz"}
    ];
  });

  beforeEach(() => {
      global.fetch.resetMocks();
  });

  it('render', done => {
    global.fetch.mockResponse(JSON.stringify(["belarus", "russia", "us", "uk"].map(item => game(item))));
    let index = createComponentWithIntl(<Index/>);
    setImmediate(() => {
      expect(index).toMatchSnapshot('Index');
      let headers = new Headers();
      headers.append("X-CSRFTOKEN", "csrf");
      expect(global.fetch).toHaveBeenCalledWith('/index/scroll/puzzle/?limit=24&ids=', {"credentials": "same-origin", "headers": headers});
      done();
    });
  });
});
