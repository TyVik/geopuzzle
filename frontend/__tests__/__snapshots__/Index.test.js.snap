// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallow <Index /> components render: Index 1`] = `
Array [
  <div
    className="nav nav-tabs nav-fill"
    onKeyDown={[Function]}
    role="tablist"
  >
    <div
      className="nav-item"
    >
      <a
        aria-controls="indexTab-tabpane-puzzle"
        aria-selected={true}
        className="puzzle-tab nav-link active"
        data-rr-ui-event-key="puzzle"
        disabled={false}
        id="indexTab-tab-puzzle"
        onClick={[Function]}
        role="tab"
      >
        Puzzle
      </a>
    </div>
    <div
      className="nav-item"
    >
      <a
        className="quiz-tab nav-link"
        data-rr-ui-event-key="quiz"
        disabled={false}
        id="indexTab-tab-quiz"
        onClick={[Function]}
        role="tab"
        tabIndex={-1}
      >
        Quiz
      </a>
    </div>
  </div>,
  <div
    className="tab-content"
  >
    <div
      aria-labelledby="indexTab-tab-puzzle"
      className="fade puzzle-tab tab-pane active show"
      id="indexTab-tabpane-puzzle"
      role="tabpanel"
    >
      <div
        className="p-3 d-flex"
      >
        <video
          autoPlay={true}
          className="flex-shrink-0"
          loop={true}
          muted={true}
          playsInline={true}
        >
          <source
            src="/static/images/puzzle.mp4"
            type="video/mp4"
          />
        </video>
        <div
          className="blockquote flex-grow-1 ms-3"
        >
          <p>
            index.description.puzzle
          </p>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-12 my-4 item-container col-md-6 col-sm-6"
        >
          <a
            href="/puzzle/hard/"
          >
            <img
              alt="hard"
              className="img-fluid rounded"
              src="hard.jpg"
            />
          </a>
          <div>
            hard
          </div>
        </div>
        <div
          className="col-12 my-4 item-container col-md-6 col-sm-6"
        >
          <a
            href="/puzzle/easy/"
          >
            <img
              alt="easy"
              className="img-fluid rounded"
              src="easy.jpg"
            />
          </a>
          <div>
            easy
          </div>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-6 my-4 item-container col-md-3 col-sm-3"
        >
          <a
            href="/puzzle/europe/"
          >
            <img
              alt="europe"
              className="img-fluid rounded"
              src="europe.jpg"
            />
          </a>
          <div>
            europe
          </div>
        </div>
        <div
          className="col-6 my-4 item-container col-md-3 col-sm-3"
        >
          <a
            href="/puzzle/america/"
          >
            <img
              alt="america"
              className="img-fluid rounded"
              src="america.jpg"
            />
          </a>
          <div>
            america
          </div>
        </div>
        <div
          className="col-6 my-4 item-container col-md-3 col-sm-3"
        >
          <a
            href="/puzzle/asia/"
          >
            <img
              alt="asia"
              className="img-fluid rounded"
              src="asia.jpg"
            />
          </a>
          <div>
            asia
          </div>
        </div>
        <div
          className="col-6 my-4 item-container col-md-3 col-sm-3"
        >
          <a
            href="/puzzle/africa/"
          >
            <img
              alt="africa"
              className="img-fluid rounded"
              src="africa.jpg"
            />
          </a>
          <div>
            africa
          </div>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-6 my-2 item-container col-md-2 col-sm-3"
        >
          <a
            href="/puzzle/belarus/"
          >
            <img
              alt="belarus"
              className="img-fluid rounded"
              src="belarus.jpg"
            />
          </a>
          <div>
            belarus
          </div>
        </div>
        <div
          className="col-6 my-2 item-container col-md-2 col-sm-3"
        >
          <a
            href="/puzzle/russia/"
          >
            <img
              alt="russia"
              className="img-fluid rounded"
              src="russia.jpg"
            />
          </a>
          <div>
            russia
          </div>
        </div>
        <div
          className="col-6 my-2 item-container col-md-2 col-sm-3"
        >
          <a
            href="/puzzle/us/"
          >
            <img
              alt="us"
              className="img-fluid rounded"
              src="us.jpg"
            />
          </a>
          <div>
            us
          </div>
        </div>
        <div
          className="col-6 my-2 item-container col-md-2 col-sm-3"
        >
          <a
            href="/puzzle/uk/"
          >
            <img
              alt="uk"
              className="img-fluid rounded"
              src="uk.jpg"
            />
          </a>
          <div>
            uk
          </div>
        </div>
      </div>
      <div
        className="row"
      />
    </div>
  </div>,
]
`;
